### PR TITLE
[Doris On ES]fix bug of value in like not translate correcttly

### PR DESCRIPTION
**Pull request by BabySid**
case is desc in #3576 

**Why this case happened**
In current implement, translation into dsl only if it is not the first charactor.
Thus, when sql is write like '%abc', translation would not run.

**How fixed**
- Now, translation will trigger with charactor '?' or '*'
- if it is the first charactor, translate directly
- else,  check the preceding char is escaped or not to determin translation or not

### CODE REVIEW

File: be/src/exec/es/es_query_builder.cpp [hunk 1]
```diff
@@ -113,12 +113,17 @@ WildCardQueryBuilder::WildCardQueryBuilder(const ExtLikePredicate& like_predicat
...
     //      abc\\_123 ===> abc\\_123
     //      abc\\%123 ===> abc\\%123
     // NOTE. user must input sql like 'abc\\_123' or 'abc\\%ykz'
     for (int i = 0; i< _like_value.size(); i++) {
-        if ((_like_value[i] == '_' || _like_value[i]== '%') && i > 0) {
-                if (_like_value[i-1] != '\\' ) {
+        if (_like_value[i] == '_' || _like_value[i]== '%') {
+                if (i == 0) {
+                    _like_value[i] = (_like_value[i] == '_') ? '?' : '*';
+                }else if (_like_value[i-1] != '\\' ) {
```
Comment (wuyunfeng) [inline be/src/exec/es/es_query_builder.cpp:126]:
```suggestion
                } else if (_like_value[i-1] != '\\' ) {
```

**Review by wuyunfeng — DISMISSED** [review]
LGTM

**Comment by blackfox1983** [comment]
@imay pls have a look?

File: be/src/exec/es/es_query_builder.cpp [hunk 2]
```diff
@@ -113,12 +113,17 @@ WildCardQueryBuilder::WildCardQueryBuilder(const ExtLikePredicate& like_predicat
...
+    //      %abc123  ===> *abc123
+    //      _abc123  ===> ?abc123
+    //      \\_abc1  ===> \\_abc1
     //      abc\\_123 ===> abc\\_123
     //      abc\\%123 ===> abc\\%123
     // NOTE. user must input sql like 'abc\\_123' or 'abc\\%ykz'
     for (int i = 0; i< _like_value.size(); i++) {
-        if ((_like_value[i] == '_' || _like_value[i]== '%') && i > 0) {
-                if (_like_value[i-1] != '\\' ) {
+        if (_like_value[i] == '_' || _like_value[i]== '%') {
```
Comment (imay) [inline be/src/exec/es/es_query_builder.cpp:123]:
```suggestion
        if (_like_value[i] == '_' || _like_value[i] == '%') {
```

File: be/src/exec/es/es_query_builder.cpp [hunk 3]
```diff
@@ -113,12 +113,17 @@ WildCardQueryBuilder::WildCardQueryBuilder(const ExtLikePredicate& like_predicat
...
+                if (i == 0) {
+                    _like_value[i] = (_like_value[i] == '_') ? '?' : '*';
+                } else if (_like_value[i-1] != '\\' ) {
```
Comment (imay) [inline be/src/exec/es/es_query_builder.cpp:126]:
```suggestion
                } else if (_like_value[i - 1] != '\\' ) {
```

**Review by imay — APPROVED** [review]
LGTM

**Comment by blackfox1983** [comment]
reopened https://github.com/apache/incubator-doris/pull/3602

**Pull request closed without merging**